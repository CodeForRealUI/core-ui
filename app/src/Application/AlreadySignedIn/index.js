import Loadable from 'react-loadable';

export default Loadable({
  loader: () => import('./AlreadySignedIn'),
  loading: () => null,
});
