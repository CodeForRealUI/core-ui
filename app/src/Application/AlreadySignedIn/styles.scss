@import '~sass/variables';
@import '~sass/colors';
@import '~sass/mixins';

html, body {
    background: $sign-in-background;
    width: 100%;
    height: 100%;
    margin: 0;
}
.already-signed-in-container {
    @include center-item();

    .already-signed-in-box {
        text-align: center;
        padding: 40px;

        .continue-to-dashboard-button {
            @include button
        }
    }
  
}