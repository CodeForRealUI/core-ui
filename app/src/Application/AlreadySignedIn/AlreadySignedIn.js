import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Paper } from 'material-ui';
import { Link, Redirect } from 'react-router-dom';

import { signoutRequest } from '~/data/actions/signout';
import './styles.scss';

const AlreadySignedIn = ({ handleSignoutRequest }) => (
  <div className="already-signed-in-container">
    <Paper className="already-signed-in-box" elevation={24}>
      <h2>You are already signed in</h2>
      <p><button
        className="continue-to-dashboard-button"
      >

        <Link className="continue-to-dashboard-link" to="/dashboard">
         Continue to Dashboard
      </Link>
      </button>
      </p>
      <button onClick={handleSignoutRequest}>Sign Out Instead</button>
    </Paper>
  </div>
);
AlreadySignedIn.propTypes = {
  handleSignoutRequest: PropTypes.func.isRequired,
};

export default connect(null, dispatch => ({
  handleSignoutRequest: () => dispatch(signoutRequest()),
}))(AlreadySignedIn);
