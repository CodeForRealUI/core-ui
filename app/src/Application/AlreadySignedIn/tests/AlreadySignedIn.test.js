import React from 'react';
import renderer from 'react-test-renderer';

import { AlreadySignedIn } from '../AlreadySignedIn';

jest.mock('react-router-dom', () => ({ Link: 'Link' }));

describe('Component AlreadySignedIn', () => {
  const props = {
    handleSignoutRequest: () => {},
    isMissingRole: true,
  };
  it('should remain consistent if no change was intended', () => {
    const tree = renderer.create(<AlreadySignedIn {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
