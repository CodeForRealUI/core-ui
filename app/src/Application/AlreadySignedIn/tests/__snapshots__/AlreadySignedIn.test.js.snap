// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component AlreadySignedIn should remain consistent if no change was intended 1`] = `
<div
  className="already-signed-in-container"
>
  <div
    className="MuiPaper-root-1 MuiPaper-elevation24-27 MuiPaper-rounded-2 already-signed-in-box"
  >
    <h1
      className="header"
    >
      You are already signed in
    </h1>
    <button
      className="continue-to-dashboard-button"
    >
      <Link
        className="continue-to-dashboard-link"
        to="/verify-role"
      >
        Continue to Dashboard
      </Link>
    </button>
  </div>
  <Link
    className="sign-out-link"
    onClick={[Function]}
    to="/sign-in"
  >
    Sign Out Instead
  </Link>
</div>
`;
